### PR TITLE
Fixes error with Mean Characters Per Word with Series of Nones

### LINKED ISSUE #100: Mean Characters Per Word returns error on array of None
Mean Characters Per Word returns error on array of None

```python
from nlp_primitives import mean_characters_per_word

mean_characters_per_word = mean_characters_per_word.get_function()

mean_characters_per_word(pd.Series([None, None, None])
```

### PR DESCRIPTION
**Pull request by dvreed77**
Fixes error with Mean Characters Per Word with Series of Nones

- fixes #100

### CODE REVIEW

**Comment by dvreed77** [comment]
This method also drastically improves performance by over 80x.

CleanShot 2022-02-17 at 10 21 03
CleanShot 2022-02-17 at 10 07 44

File: nlp_primitives/tests/test_mean_characters_per_word.py [hunk 1]
```diff
@@ -41,7 +41,31 @@ def test_nans(self):
                        '',
                        'third line'])
         primitive_func = self.primitive().get_function()
-        answers = pd.Series([np.nan, np.nan, 4.5])
+        answers = pd.Series([np.nan, 0, 4.5])
+        pd.testing.assert_series_equal(primitive_func(x), answers, check_names=False)
+
+    def test_all_nans(self):
```
Comment (thehomebrewnerd) [inline nlp_primitives/tests/test_mean_characters_per_word.py:47]: Could we test this with `pd.NA` instead of `np.nan` as a `NaturalLanguage` column with use the pandas `string` dtype which should result in `pd.NA` values being a passed to the primitive function.

Could also maybe parameterize a new test replacing `test_all_nans` and `test_none` to check with `None`, `pd.NA` and `np.nan` as they should all produce the same result.

File: nlp_primitives/mean_characters_per_word.py [hunk 2]
```diff
@@ -29,17 +46,8 @@ class MeanCharactersPerWord(TransformPrimitive):
...
-            x = x.str.replace(end_of_sentence_punct, ' ')
-            # build DF of split words, and calculate avg length
-            df = x.str.split(expand=True)
-            df = df.reset_index()
-            df = df.melt(id_vars='index', var_name='word_index', value_name='word')
-            df['n_characters'] = df['word'].str.len()
-            results = df.groupby(by=['index']).mean(numeric_only=False)['n_characters']
-            return results
+        def mean_characters_per_word(series):
+            series = series.reset_index(drop=True)
```
Comment (thehomebrewnerd) [inline nlp_primitives/mean_characters_per_word.py:50]: Do we need to do this `reset_index` call?

**Review by thehomebrewnerd — COMMENTED** [review]
Looks pretty good, just a couple small suggestions/questions. Also should add an entry to `changelog.rst` in the `docs/source` folder.

File: nlp_primitives/mean_characters_per_word.py (hunk 2 above)
Comment (dvreed77) [reply nlp_primitives/mean_characters_per_word.py:50]: I don't think so. It was already there, and I wasn't sure if it was capturing something I'm not familiar with

**Review by thehomebrewnerd — APPROVED** [review]
LGTM, assuming tests pass, of course.
